### DISCUSSION TITLE
How Can I customize/add some docker run arguments on community containers

### DISCUSSION THREAD

**Discussion by luzfcb** [nextcloud/all-in-one#3847] [Questions] [answered]
### How Can I customize/add some docker run arguments on community containers?

In this case, I would like to include one or more entries of the `--dns` argument on the docker run command to make the container query some extra dns servers

https://docs.docker.com/network/#dns-services 

Another question I have is where `memories.json` (https://github.com/nextcloud/all-in-one/blob/main/community-containers/memories/memories.json) is persisted on my local installation of Nextcloud AIO and if it is possible for me to customize it in some way (the current documentation is not very clear on this specific topic)

##### A bit more context: 

I have Nextcloud AIO running on a non-public server (I don't have a real external IP address).

To get it working on a non-public network, I had to do a few things

1 - I have a domain, let's call it `nextcloud.mydomain.com`
2 - I have a VPS server and nextcloud.mydomain.com points to it.
3 - I have a server with Nextcloud AIO running inside my home, which does not have a public IP.
4 - To access the Nextcloud AIO server via the internet, I have a VPN configured on the VPS, and Caddy reverse proxying the Nextcloud AIO Apache server host within my VPN. (this works great)
5 - I have a DNS server running on my home Internet Gateway.
6 - When I'm using Nextcloud inside my home, I avoid data traffic leaving my home, going to my VPS server and returning via VPN to my home by intercepting the DNS query to nextcloud.mydomain.com and making it point to an internal IP address instead of the public IP address.
7 - Inside my home, I have a Caddy server doing a reverse proxy for the Nextcloud AIO Apache server host (this works great).

I am trying to configure Memories community container, however, the  `nextcloud-aio-memories` container is getting the error

```bash
Failed to fetch https://nextcloud.mydomain.com/index.php/apps/memories/static/go-vod?arch=x86_64
Are you sure the host is reachable and running Memories v6+?
Retrying in 10 seconds
```

If I access the URL described in the log outside of docker, everything works as expected

```
$ wget https://nextcloud.mydomain.com/index.php/apps/memories/static/go-vod?arch=x86_64 -O go-vod
--2023-11-30 13:50:32--  https://nextcloud.mydomain.com/index.php/apps/memories/static/go-vod?arch=x86_64
Resolving nextcloud.mydomain.com (nextcloud.mydomain.com)... 192.168.50.40
Connecting to nextcloud.mydomain.com (nextcloud.mydomain.com)|192.168.50.40|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: unspecified [text/html]
Saving to: ‘go-vod’

go-vod                          [ <=>                                       ]   4,83M  --.-KB/s    in 0,04s   

2023-11-30 13:50:32 (111 MB/s) - ‘go-vod’ saved [5066752]

```

Additional notes: I noticed that the `nextcloud-aio-memories` container does not have the `arp`, `rarp`, `ip`, `ping`, `tracepath` or `traceroute` commands installed. Well, however, I don't know if they are really necessary, I'm just reporting it because it made my debugging process a little more difficult.

**Comment by szaimen** [comment, marked as answer]
> How Can I customize/add some docker run arguments on community containers

You cant.

But you can set the dns server globally using the daemon.json file. See https://dockerlabs.collabnix.com/intermediate/networking/Configuring_DNS.html
